### ISSUE TITLE
EmptyLineAfterGuardClause should accept commented guard clauses

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14702]
## Steps to reproduce the problem

```ruby
4.times do |bear|
  next puts 'Papa' if bear == 1
  next puts 'Mama' if bear == 2 
  # next puts 'Little' if bear == 3

  puts 'Goldilocks'
end
```

## Expected behavior

- Commented guard clauses: No cop
- Other comments: Cop

## Actual behavior

Nothing happens if we comment Papa Bear or Mama Bear, but if we comment out Little Bear, Mama Bear goes `EmptyLineAfterGuardClause`. This is something of an annoyance, and you have to shuffle the bears to make the Park Ranger happy. However, comments that are *not* temporarily commented guard clauses should arguably still have their stockings coaled by Santa Claus. (Puns are half the reward of writing these issues...)

## RuboCop version

```
System Version: macOS 26.1 (25B78)
Model Identifier: MacBookAir10,1
ruby 3.4.7 (2025-10-08 revision 7a5688e2a2) +PRISM [arm64-darwin24]
rubocop     2025-12-02 9338710b3

```

**Comment by dvandersluis** [comment]
Comments aren't parsed so RuboCop doesn't know that the comment would be a guard clause if not commented out.

Currently the cop allows RuboCop directive and `nocov` comments to follow guard clauses. We could theoretically extend that with a configuration to ignore any comment lines.

**Comment by forthrin** [comment]
@dvandersluis: Do you agree with LLM and I that this is the way to go?

---

RuboCop can access comments through `processed_source.comments`, but comments are not part of the normal AST, and distinguishing commented-out code from ordinary prose requires heuristic parsing.

### Recommended approach

Add a configuration option to `Layout/EmptyLineAfterGuardClause`:

```yaml
Layout/EmptyLineAfterGuardClause:
  AllowComments: true
```

- Permits any comment immediately after a guard clause.
- Continues enforcing the empty line when the next line is executable code.
- Preserves the existing exceptions for RuboCop directives and `nocov`.

This is deterministic, cheap, and avoids false positives.

### Cost-benefit assessment

| Approach | Cost | Accuracy | Recommendation |
|---|---:|---:|---|
| Ignore all comments after guard clauses | Low | High | Best default feature |
| Configurable comment allowance | Low | High | Recommended |
| Parse comments as Ruby | High | Low–Medium | Avoid |
| Build a commented-code classifier | Very high | Medium | Not worthwhile |

A configurable `AllowComments` option solves the reported annoyance while preserving predictable cop behavior.
